### PR TITLE
SCRUM-24-Realisation-method-and-test-getProducts and connect swagger

### PR DESCRIPTION
**Pull request by FartushkoY**

### CODE REVIEW

File: src/main/java/de/telran/onlineshopforhomeandgarden1/mapper/ProductMapper.java [hunk 1]
```diff
@@ -14,4 +17,19 @@ public interface ProductMapper {
     ProductDto entityToDto(Product product);
 
     List<ProductDto> entityListToDto(List<Product> products);
+
+    //    RequestDto
+    Product RequestDtoToEntity(ProductRequestDto productDto);
```
Comment (VoytovychM) [inline src/main/java/de/telran/onlineshopforhomeandgarden1/mapper/ProductMapper.java]: please do the to lowercase in the name of method

File: src/test/java/de/telran/onlineshopforhomeandgarden1/service/ProductServiceTest.java [hunk 2]
```diff
@@ -51,4 +55,27 @@ public void getProductByIdIncorrectTest() {
...
+        Boolean hasDiscount = null;
+        Integer minPrice = null;
+        Integer maxPrice = null;
+        List<Product> productList = Arrays.asList(product);
+        Page<Product> products = new PageImpl<>(productList);
+//        Page<Product> products = new PageImpl<>(Collections.singletonList(product));
+        ProductWithDiscountPriceResponseDto responseDto = new ProductWithDiscountPriceResponseDto();
+//        productMapper = Mockito.mock(ProductMapper.class);
+
+        Mockito.when(repository.getAllWithFilters(categoryId, null, false, BigDecimal.valueOf(0), BigDecimal.valueOf(Integer.MAX_VALUE), pageable))
```
Comment (VoytovychM) [inline src/test/java/de/telran/onlineshopforhomeandgarden1/service/ProductServiceTest.java]: Mock expects  specific arguments and the service passes other arguments for example hasdiscount you pass as null and the service initialises them as false.

**Review by VoytovychM — CHANGES_REQUESTED** [review]

File: src/main/java/de/telran/onlineshopforhomeandgarden1/mapper/ProductMapper.java (hunk 1 above)
Comment (FartushkoY) [reply src/main/java/de/telran/onlineshopforhomeandgarden1/mapper/ProductMapper.java]: Исправила

File: src/test/java/de/telran/onlineshopforhomeandgarden1/service/ProductServiceTest.java (hunk 2 above)
Comment (FartushkoY) [reply src/test/java/de/telran/onlineshopforhomeandgarden1/service/ProductServiceTest.java]: Все работает. Спасибо!

**Review by VoytovychM — APPROVED** [review]

**Pull request merged**